python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/dlt/asset_decorator.py

```diff
@@ -14,13 +14,57 @@
 from .translator import DagsterDltTranslator
 
 
+def build_dlt_asset_specs(
+    dlt_source: DltSource,
+    dlt_pipeline: Pipeline,
+    dagster_dlt_translator: Optional[DagsterDltTranslator] = None,
+) -> List[AssetSpec]:
+    """Build a list of asset specs from a dlt source and pipeline.
+
+    Args:
+        dlt_source (DltSource): dlt source object
+        dlt_pipeline (Pipeline): dlt pipeline object
+        dagster_dlt_translator (Optional[DagsterDltTranslator]): Allows customizing how to
+            map dlt project to asset keys and asset metadata.
+
+    Returns:
+        List[AssetSpec] list of asset specs from dlt source and pipeline
+
+    """
+    dagster_dlt_translator = dagster_dlt_translator or DagsterDltTranslator()
+    destination_type = dlt_pipeline.destination.destination_name
+    return [
+        AssetSpec(
+            key=dagster_dlt_translator.get_asset_key(dlt_source_resource),
+            auto_materialize_policy=dagster_dlt_translator.get_auto_materialize_policy(
+                dlt_source_resource
+            ),
+            deps=dagster_dlt_translator.get_deps_asset_keys(dlt_source_resource),
+            description=dagster_dlt_translator.get_description(dlt_source_resource),
+            group_name=dagster_dlt_translator.get_group_name(dlt_source_resource),
+            metadata={
+                META_KEY_SOURCE: dlt_source,
+                META_KEY_PIPELINE: dlt_pipeline,
+                META_KEY_TRANSLATOR: dagster_dlt_translator,
+                **dagster_dlt_translator.get_metadata(dlt_source_resource),
+            },
```

Comment:
To ensure that the internal metadata keys don't actually get overwritten

```suggestion
            metadata={
                **dagster_dlt_translator.get_metadata(dlt_source_resource),
                META_KEY_SOURCE: dlt_source,
                META_KEY_PIPELINE: dlt_pipeline,
                META_KEY_TRANSLATOR: dagster_dlt_translator,             
            },
```